File: 2018/March/March-11/March-11-2018.rb
#***************************
#*  191. Number of 1 Bits  *
#***************************

#O(1) solution [Because n is a 32-bit integer, and that's what is recursed through]
def hamming_weight(n)
    count = 0
    for i in 0..31
        if (n & 1) == 1
            count = count + 1
        end
        n = n >> 1
    end
    
    return count
end

File: 2018/March/March-11/March-11-2018.js
//***************************
//*  191. Number of 1 Bits  *
//***************************

//O(1) solution [Because n is a 32-bit integer, and that's what is looped through]
var hammingWeight = function(n)
{
    var count = 0;
    var i = 0;
    var bits = 0;
    
    while (i < 32)
    {
        bits = n & 1;
        if (bits == 1)
        {
            count += 1;
        };
        
        i += 1;
        n >>= 1;
    };
    
    return count;
};